service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/RepresentativesInbox.java

```diff
@@ -47,11 +56,44 @@
                 .collect(Collectors.toCollection(LinkedHashSet::new))
         );
 
-
         return emails;
     }
 
-    private boolean shouldSend(RepresentativeServingPreferences preference, Element<? extends WithSolicitor> element) {
+    public Set<?> getNonSelectedRespondentsRecipients(
+        RepresentativeServingPreferences servingPreferences,
+        CaseData caseData,
+        List<Element<Respondent>> respondentsSelected,
+        Function<Element<Representative>, ?> mapperFunction) {
+
+        Set<UUID> representativeIds = caseData.getAllRespondents().stream()
+            .flatMap(respondentElement -> respondentElement.getValue().getRepresentedBy()
```

Comment:
adding a map before the flat map may make it easier to read (Element::getValue)

Reply:
updated to use `unwrapElements()`